FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -619,4 +625,57 @@ - (FIRConfigUpdateListenerRegistration *)addOnConfigUpdateListener:
   return [self->_configRealtime addConfigUpdateListener:listener];
 }
 
+#pragma mark - Rollout
+
+- (void)addRemoteConfigInteropSubscriber:(id<FIRRolloutsStateSubscriber>)subscriber {
+  [[NSNotificationCenter defaultCenter]
+      addObserverForName:RolloutsStateDidChangeNotificationName
+                  object:self
+                   queue:nil
+              usingBlock:^(NSNotification *_Nonnull notification) {
+                FIRRolloutsState *rolloutsState =
+                    notification.userInfo[RolloutsStateDidChangeNotificationName];
+                [subscriber rolloutsStateDidChange:rolloutsState];
+              }];
+  // Send active rollout metadata stored in persistence while app launched if there is activeConfig
+  NSString *FQNamespace = [self fullyQualifiedNamespace:_FIRNamespace];
+  NSDictionary<NSString *, NSDictionary *> *activeConfig = self->_configContent.activeConfig;
+  if (activeConfig[FQNamespace].count > 0) {
+    [self notifyRolloutsStateChange:self->_configContent.activeRolloutMetadata
+                      versionNumber:self->_settings.lastActiveTemplateVersion];
+  }
+}
+
+- (void)notifyRolloutsStateChange:(NSArray<NSDictionary *> *)rolloutMetadata
+                    versionNumber:(NSString *)versionNumber {
+  NSMutableArray<FIRRolloutAssignment *> *rolloutsAssignments = [[NSMutableArray alloc] init];
+  NSString *FQNamespace = [self fullyQualifiedNamespace:_FIRNamespace];
+  for (NSDictionary *metadata in rolloutMetadata) {
+    NSString *rolloutId = metadata[RCNFetchResponseKeyRolloutID];
+    NSString *variantID = metadata[RCNFetchResponseKeyVariantID];
+    NSArray<NSString *> *affectedParameterKeys = metadata[RCNFetchResponseKeyAffectedParameterKeys];
+    if (rolloutId && variantID && affectedParameterKeys) {
+      for (NSString *key in affectedParameterKeys) {
+        FIRRemoteConfigValue *value = self->_configContent.activeConfig[FQNamespace][key];
+        if (value) {
```

Comment:
Does allow an empty string value? (we want to allow that since it's a valid value from the backend)

Other note here is that we want to fall back to the in-app default value or the static default if there isn't a value present in the active config (i.e. [following the SDK's `get(param)` logic](https://github.com/firebase/firebase-ios-sdk/blob/7b9125a2e99529e48a2e19dd6892c950e3dda58d/FirebaseRemoteConfig/Sources/FIRRemoteConfig.m#L405-L413))

Reply:
Yes, empty string will return true when do if check on it and nil will return false. 
Add in-app default value logic if value is nil.

Reply:
Maybe I'm missing it - can you point me to where the in-app default fallback logic is?